taca/analysis/analysis_nanopore.py

```diff
@@ -48,221 +48,191 @@ def send_error_mail(run_name, error: BaseException):
+            raise AssertionError(f"{run.run_name}: Anglerfish failed.")
+    else:
+        raise AssertionError("Unexpected Anglerfish exit code.")
 
-            logger.info(f"{ont_qc_run.run_name}: Fetching Anglerfish samplesheet...")
-
-            if not ont_qc_run.fetch_anglerfish_samplesheet():
-                raise WaitForRun("Could not find Anglerfish sample sheet, skipping.")
-
-            logger.info(f"{ont_qc_run.run_name}: Starting Anglerfish...")
-            ont_qc_run.run_anglerfish()
-            raise WaitForRun("Anglerfish has been started, skipping.")
-
-        # Anglerfish run
-        elif isinstance(anglerfish_exit_code, int):
-            if anglerfish_exit_code == 0:
-                logger.info(
-                    f"{ont_qc_run.run_name}: Anglerfish has finished successfully, continuing."
-                )
-            elif anglerfish_exit_code > 0:
-                logger.error(
-                    f"{ont_qc_run.run_name}: Anglerfish has failed, throwing error."
-                )
-                raise AssertionError(f"{ont_qc_run.run_name}: Anglerfish failed.")
-        else:
-            raise AssertionError("Unexpected Anglerfish exit code.")
-
-    # Check transfer status
-    if ont_qc_run.is_transferred():
+    # Copy metadata
+    logger.info(f"{run.run_name}: Copying metadata...")
+    run.copy_metadata()
+
+    # Transfer to analysis server
+    if run.transfer_status == "not started":
+        logger.info(f"{run.run_name}: Starting transfer...")
+        run.transfer()
+    elif run.transfer_status == "ongoing":
+        raise WaitForRun(f"{run.run_name}: Transfer is ongoing, skipping.")
+    elif run.transfer_status == "rsync done":
+        logger.info(f"{run.run_name}: Transfer complete. Archiving...")
+        run.update_transfer_log()
+        run.remove_transfer_indicator()
+        run.archive_run()
+    elif run.transfer_status == "rsync failed":
+        raise AssertionError(f"{run.run_name}: Transfer failed, please investigate.")
+    elif run.transfer_status == "transferred":
         logger.warning(
-            f"{ont_qc_run.run_name}: Run is already logged as transferred, skipping."
+            f"{run.run_name}: Run is already logged as transferred, skipping."
         )
         raise WaitForRun("Run is already logged as transferred.")
-
-    logger.info(f"{ont_qc_run.run_name}: Processing transfer...")
-
-    # Copy metadata
-    logger.info(f"{ont_qc_run.run_name}: Copying metadata...")
-    ont_qc_run.copy_metadata()
-
-    # Transfer run
-    logger.info(f"{ont_qc_run.run_name}: Transferring to cluster...")
-    ont_qc_run.transfer_run()
-
-    # Update transfer log
-    logger.info(f"{ont_qc_run.run_name}: Updating transfer log...")
-    ont_qc_run.update_transfer_log()
-
-    # Archive run
-    logger.info(f"{ont_qc_run.run_name}: Archiving run...")
-    ont_qc_run.archive_run()
+    else:
+        raise AssertionError(
```

Comment:
Since process_user_run checks the same thing, maybe this could be a separate function so that we don't have to change it in two places if the logic needs to be updated?